keps/sig-auth/2579-psp-replacement/README.md

```diff
@@ -425,41 +418,25 @@ against audit & warn policies, independent of which fields are being modified.
 
 #### Ephemeral Containers
 
-In the initial implementation, ephemeral containers will be subject to the same policy restrictions,
+Ephemeral containers will be subject to the same policy restrictions,
 and adding or updating ephemeral containers will require a full policy check.
-
-<<[UNRESOLVED]>>
-
-_Non-blocking for alpha. This should be resolved for beta._
-
-Once ephemeral containers allow [custom security contexts], it may be desirable to run an ephemeral
-container with higher privileges for debugging purposes. For example, CAP_SYS_PTRACE is forbidden by
-the baseline policy but can be useful in debugging. We could introduce yet-another-mode-label that
-only applies enforcement to ephemeral containers (defaults to the enforce policy).
-
-[custom security contexts]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/277-ephemeral-containers#configurable-security-policy
-
-One way this could be handled under the current model is:
-1. Exempt a special username (not one that can be authenticated directly) from policy enforcement,
-   e.g. `ops:privileged-debugger`
-2. Grant the special user permission to ONLY operate on the ephemeral containers subresource (it is
-   critical that they cannot create or update pods directly).
-3. Grant (real) users that should have privileged debug capability the ability to impersonate the
-   exempt user.
-
-We could consider ways to streamline the user experience of this, for instance adding a special RBAC
-binding that exempts users when operating on the ephemeral containers subresource (e.g. an
-`escalate-privilege` verb on the ephemeral containers subresource).
-
-<<[/UNRESOLVED]>>
+This means that an existing pod which is not valid according to the current
```

Comment:
> Since ephemeral containers cannot be removed, it sounds like it's possible for a pod to fail the current enforce level even if the ephemeral container is no longer running, blocking future updates that would otherwise be allowed at this level. Is that desirable?

I think so? Some of the changes an escalated ephemeralContainer could make could persist even after the ephemeral container is no longer running. Presence of an ephemeral container in a pod that doesn't fit within the current enforce level seems like it should be rejected, even if it is not running.

Reply:
There are basically 3 operations that would trigger a pod to be reevaluated, and potentially block the update:

1. Change the seccomp annotation - deprecated behavior that will be removed soon.
2. Updating the container image - goes against best practices
3. Adding an ephemeral container

So, if an exempt user adds a privileged ephemeral container, the pod enters a "violating" state, and non-exempt users can no longer perform any of those updates. This seems reasonable to me.

Reply:
> One way this could be handled under the current model is:
> 1. Exempt a special username (not one that can be authenticated directly) from policy enforcement,
>    e.g. `ops:privileged-debugger`
> 2. Grant the special user permission to ONLY operate on the ephemeral containers subresource (it is
>    critical that they cannot create or update pods directly).
> 3. Grant (real) users that should have privileged debug capability the ability to impersonate the
>    exempt user.

I still think it's useful to document this workflow somewhere. This behavior would be a bit more intuitive if we had group exemptions (impersonate the breakglass group), but the workflow doesn't change that much as long as RBAC bindings are mostly granted to groups.